src/indexer/db_con.rs

```diff
@@ -0,0 +1,31 @@
+use sqlx::postgres::PgPoolOptions;
+use sqlx::{Pool, Postgres};
+use urlencoding::encode;
+
+pub async fn setup_db_connection(database_url: &str) -> Result<Pool<Postgres>, anyhow::Error> {
+  let encoded_database_url = encode_password_in_url(database_url);
+  let pool = PgPoolOptions::new()
+    .max_connections(5)
```

Comment:
This should probably be passed in `setup_db_connection(database_url: &str, connections: u8)` so that each consumer/writer service can set a number that makes sense for it.

I presume `5` is too many in most cases, e.g the publisher should have what, like 2-4? And the consumers probably something like `2` each?

```suggestion
    .max_connections(connections)
```

Reply:
Overall the function can be refined as this to reduce boilerplate.
```rs
use sqlx::postgres::PgPoolOptions;
use sqlx::{Pool, Postgres, Result};
use urlencoding::encode;

pub async fn setup_db_connection(url: &str, max_conn: u32) -> Result<Pool<Postgres>> {
  PgPoolOptions::new()
    .max_connections(max_conn)
    .connect(&encode_password_in_url(url))
    .await
}
```